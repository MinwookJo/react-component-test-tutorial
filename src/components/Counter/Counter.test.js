import React from 'react';
import renderer from 'react-test-renderer';
import Counter from './Counter';

describe('Counter Test', () => {
    let component = null;

    it('render correctly', () => {
        component = renderer.create(<Counter/>);
    });

    it('match snapshot', () => {
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('increase correctly', () => {
        const counterInstance = component.getInstance();
        counterInstance.onIncrease();
        expect(counterInstance.state.value).toBe(2);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('decrease correctly', () => {
        const counterInstance = component.getInstance();
        counterInstance.onDecrease();
        expect(counterInstance.state.value).toBe(1);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});