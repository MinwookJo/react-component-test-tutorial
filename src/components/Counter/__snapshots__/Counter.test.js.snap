// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Test decrease correctly 1`] = `
<div>
  <h1>
    카운터!
  </h1>
  <h2>
    1
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;

exports[`Counter Test increase correctly 1`] = `
<div>
  <h1>
    카운터!
  </h1>
  <h2>
    2
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;

exports[`Counter Test match snapshot 1`] = `
<div>
  <h1>
    카운터!
  </h1>
  <h2>
    1
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
